### PR TITLE
Menu button and popover can have same width

### LINKED ISSUE #4858: [Components][Menu] Should be able to make the popover's width equivalent to the Trigger button
### Provide a general summary of the feature here

Was playing with [Menu](https://react-spectrum.adobe.com/react-aria/Menu.html) component and it looks good to be in terms of the overall composition and structure, but one thing that I missed was to be able to make the Menu's popover's width equal to the Menu's trigger Button.

I couldn't find a way to do it without handling the width manually for both Popover and Button.

There should be a prop or some other way of doing this.

menu

### 🤔 Expected Behavior?

Ideally, there can be a prop or some way of doing this.

### 😯 Current Behavior

Right now, it's not possible to do this without setting and handling the widths manually.

### 💁 Possible Solution

_No response_

### 🔦 Context

Most of the times, we want the menus to be able to have equal width of their trigger buttons, especially on Mobile devices.

### 💻 Examples

_No response_

### 🧢 Your Company/Team

_No response_

### 🕷 Tracking Issue

_No response_

### PR DESCRIPTION
**Pull request by ktabors**
Closes #4858

RAC: Adding --trigger-width support to Menu

When coping I left the resizer logic in case someone tries to change the button contents. We can remove that if we don't want to support that.

I can also remove the example story after testing, before merging.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Added a new story, "Menu trigger width example" that has this width setup.

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/react-aria-components/example/index.css [hunk 1]
```diff
@@ -12,6 +12,14 @@ html {
   background: lightgray;
 }
 
+.react-aria-Popover[data-trigger=MenuTrigger] {
+  width: var(--trigger-width);
+}
+
+.menuTriggerWidth {
+  min-width: var(--trigger-width);
```
Comment (snowystinger) [inline packages/react-aria-components/example/index.css:20]: why was the min width needed?

File: packages/react-aria-components/example/index.css [hunk 2]
```diff
@@ -3,7 +3,7 @@ html {
   font-family: system-ui;
 }
 
-.menu {
+.menu, .menuTriggerWidth {
```
Comment (snowystinger) [inline packages/react-aria-components/example/index.css:6]: why'd we add this here?

Comment (ktabors) [reply packages/react-aria-components/example/index.css:6]: For [this example](https://github.com/adobe/react-spectrum/pull/5465/files/f592d7d5862ae0b8d9215ba9fc6fdcf0d7e2ad71#diff-7ae373998263ffc95c0231c5f6b5b6538b2d3b2ae18056d8c016f5bb631e7cc5R322). I could change the example to be "menu menuTriggerWidth". I planned the Storybook example would go away before merging. I just wanted to make initial testing easier.

**Review by snowystinger — DISMISSED** [review]

File: packages/react-aria-components/docs/Menu.mdx [hunk 3]
```diff
@@ -76,7 +76,7 @@ import {MenuTrigger, Button, Popover, Menu, MenuItem} from 'react-aria-component
   box-sizing: border-box;
   overflow: auto;
   padding: 2px;
-  min-width: 150px;
+  min-width: min(150px, var(--trigger-width));
```
[the remark this answers is not in the document]
Comment (ktabors) [reply packages/react-aria-components/docs/Menu.mdx:79]: I had it there as a hit for how to do this correctly. I'll remove.

**Review by snowystinger — APPROVED** [review]

**Pull request merged**